addons/account/models/account_bank_statement.py

```diff
@@ -1,476 +1,245 @@
-        # To avoid this we sort the records by date
-        for statement in self.sorted(key=lambda s: s.date):
-            if statement.previous_statement_id.balance_end_real != statement.balance_start:
-                statement.balance_start = statement.previous_statement_id.balance_end_real
-            else:
-                # Need default value
-                statement.balance_start = statement.balance_start or 0.0
 
-    @api.depends('previous_statement_id', 'previous_statement_id.balance_end_real')
-    def _compute_ending_balance(self):
-        latest_statement = self.env['account.bank.statement'].search([('journal_id', '=', self[0].journal_id.id)], limit=1)
-        for statement in self:
-            # recompute balance_end_real in case we are in a bank journal and if we change the
-            # balance_end_real of previous statement as we don't want
-            # holes in case we add a statement in between 2 others statements.
-            # We only do this for the bank journal as we use the balance_end_real in cash
-            # journal for verification and creating cash difference entries so we don't want
-            # to recompute the value in that case
-            if statement.journal_type == 'bank':
-                # If we are on last statement and that statement already has a balance_end_real, don't change the balance_end_real
-                # Otherwise, recompute balance_end_real to prevent holes between statement.
-                if latest_statement.id and statement.id == latest_statement.id and not float_is_zero(statement.balance_end_real, precision_digits=statement.currency_id.decimal_places):
-                    statement.balance_end_real = statement.balance_end_real or 0.0
-                else:
-                    total_entry_encoding = sum([line.amount for line in statement.line_ids])
-                    statement.balance_end_real = statement.previous_statement_id.balance_end_real + total_entry_encoding
-            else:
-                # Need default value
-                statement.balance_end_real = statement.balance_end_real or 0.0
+    name = fields.Char(
+        string='Reference',
+        copy=False,
+        index=True,
+    )
 
-    @api.depends('line_ids', 'balance_start', 'line_ids.amount', 'balance_end_real')
-    def _end_balance(self):
-        for statement in self:
-            statement.total_entry_encoding = sum([line.amount for line in statement.line_ids])
-            statement.balance_end = statement.balance_start + statement.total_entry_encoding
-            statement.difference = statement.balance_end_real - statement.balance_end
+    # Used to hold the reference of the external mean that created this statement (name of imported file,
+    # reference of online synchronization...)
+    reference = fields.Char(
+        string='External Reference',
+        copy=False,
+    )
+    date = fields.Date(compute='_compute_date_index', store=True,)
```

Comment:
SO it is automatically filled, cleaner interface, fewer things to worry/make mistake for the user. Also, the default value won't be known because usually the users start from the earliest line to encode a statement, so if we set a default value based on the first line, it would be wrong most of the time.